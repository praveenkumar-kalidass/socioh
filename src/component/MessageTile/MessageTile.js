import React from 'react';
import { useWindowDimensions } from 'react-native';
import PropTypes from 'prop-types';

import { ICON } from '../../constant';
import {
  TileContainer,
  IconContainer,
  PointerIcon,
  Message,
} from './MessageTile.style';

const MessageTile = ({ isReceived, message, testID }) => {
  const { width } = useWindowDimensions();
  const pointer = isReceived ? ICON.CARET_LET : ICON.CARET_RIGHT;

  return (
    <TileContainer received={isReceived} maxWidth={width / 2}>
      <IconContainer received={isReceived}>
        <PointerIcon
          type={ICON.FONT_AWESOME_FAMILY}
          name={pointer}
          received={isReceived}
        />
      </IconContainer>
      <Message testID={testID} received={isReceived}>
        {message}
      </Message>
    </TileContainer>
  );
};

MessageTile.propTypes = {
  isReceived: PropTypes.bool,
  message: PropTypes.string.isRequired,
  testID: PropTypes.string,
};

MessageTile.defaultProps = {
  isReceived: false,
  testID: 'message',
};

export default MessageTile;
