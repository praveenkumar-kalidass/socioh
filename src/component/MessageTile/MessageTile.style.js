import styled from 'styled-components/native';
import { Icon, Text } from 'native-base';
import { COLOR, SPACING } from '../../constant';

export const TileContainer = styled.View`
  align-self: ${({ received }) => (received ? 'flex-start' : 'flex-end')};
  background-color: ${({ received }) =>
    received ? COLOR.PRIMARY : COLOR.SECONDARY};
  border-top-left-radius: ${SPACING.SPACE_5};
  border-top-right-radius: ${SPACING.SPACE_5};
  ${({ received }) =>
    received
      ? 'border-bottom-right-radius'
      : 'border-bottom-left-radius'}: ${SPACING.SPACE_5};
  margin-bottom: ${SPACING.SPACE_10};
  max-width: ${({ maxWidth }) => maxWidth}px;
  padding: ${SPACING.SPACE_10};
`;

export const IconContainer = styled.View`
  bottom: ${SPACING.SPACE_0};
  position: absolute;
  ${({ received }) => (received ? 'left' : 'right')}: -${SPACING.SPACE_10};
`;

export const PointerIcon = styled(Icon)`
  color: ${({ received }) => (received ? COLOR.PRIMARY : COLOR.SECONDARY)};
`;

export const Message = styled(Text)`
  color: ${({ received }) => (received ? COLOR.WHITE : COLOR.BLACK)};
`;
