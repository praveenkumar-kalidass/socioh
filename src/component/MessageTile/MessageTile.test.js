import React from 'react';
import { render } from '@testing-library/react-native';

import MessageTile from './index';

describe('MessageTile', () => {
  it('should match snapshot', () => {
    const container = render(<MessageTile message="Hi" />);

    expect(container).toMatchSnapshot();
  });

  it('should render styles for received message, when the message is received one', () => {
    const container = render(
      <MessageTile isReceived message="Good morning!" />,
    );

    expect(container).toMatchSnapshot();
  });
});
