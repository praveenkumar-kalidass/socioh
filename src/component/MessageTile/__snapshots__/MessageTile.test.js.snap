// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageTile should match snapshot 1`] = `
<View
  maxWidth={375}
  received={false}
  style={
    Array [
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "#feda68",
        "borderBottomLeftRadius": 5,
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "marginBottom": 10,
        "maxWidth": 375,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    received={false}
    style={
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
          "right": -10,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      received={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#feda68",
            },
          ],
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      type="FontAwesome"
    >
      
    </Text>
  </View>
  <Text
    received={false}
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "#000000",
        },
      ]
    }
    testID="message"
  >
    Hi
  </Text>
</View>
`;

exports[`MessageTile should render styles for received message, when the message is received one 1`] = `
<View
  maxWidth={375}
  received={true}
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#1abfbf",
        "borderBottomRightRadius": 5,
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "marginBottom": 10,
        "maxWidth": 375,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    received={true}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": -10,
          "position": "absolute",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      received={true}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#1abfbf",
            },
          ],
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      type="FontAwesome"
    >
      
    </Text>
  </View>
  <Text
    received={true}
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "#ffffff",
        },
      ]
    }
    testID="test_message"
  >
    Good morning!
  </Text>
</View>
`;
